corehq/apps/smsbillables/async_handlers.py

```diff
@@ -103,3 +106,46 @@ def _fmt_success(self, response):
             } for r in response]
         }, cls=LazyEncoder)
         return success
+
+
+class PublicSMSRatesAsyncHandler(BaseAsyncHandler):
+    slug = 'public_sms_rate_calc'
+    allowed_actions = 'public_rate'
+
+    @property
+    def public_rate_response(self):
+        return self.get_rate_table(self.data.get('country_code'))
+
+    @quickcache(['country_code'], timeout=24 * 60 * 60)
+    def get_rate_table(self, country_code):
+        backends = SMSBackend.view(
+            'sms/global_backends',
+            reduce=False,
+            include_docs=True,
+        ).all()
+
+        def _directed_fee(direction, backend_api_id, backend_instance_id):
+            gateway_fee = SmsGatewayFee.get_by_criteria(
+                backend_api_id,
+                direction,
+                backend_instance=backend_instance_id,
+                country_code=country_code
+            )
+            if not gateway_fee:
+                return None
+            usd_gateway_fee = gateway_fee.amount / gateway_fee.currency.rate_to_default
+            usage_fee = SmsUsageFee.get_by_criteria(direction)
+            return fmt_dollar_amount(usage_fee.amount + usd_gateway_fee)
+
+        rate_table = []
+        for backend_instance in backends:
+            backend_type = get_backend_by_class_name(backend_instance.doc_type)
```

Comment:
@sravfeyn well what we want to eventually use is:

    backend_instance.incoming_api_id or backend_instance.get_api_id()

but only when it's an incoming message (outgoing messages always only use get_api_id())

To prevent double-querying the docs, you could also just write a utility function that takes and SMSBackend and wraps it according to its doc_type without querying for it again.